[Temporary] feat: Taxonomy export menu

ChrisChV opened the pull request:
**Changes added:**

- ExportModal.
- Export menu added. **Note:** Is hidden for system-defined, because the export API doesn't allow export system-defined.
- Functions to make a call to the export API.
- Flow to export a taxonomy.
- System-defined tooltip.

**Screenshots**

![image](https://github.com/open-craft/frontend-app-course-authoring/assets/11827305/f6818337-bef7-41d7-9f0b-8c890e664f99)
![image](https://github.com/open-craft/frontend-app-course-authoring/assets/11827305/223de2c2-4595-4e1d-b6a8-a59a18b58371)

**Testing instructions**

- Verify the tooltip of the badge of a System-defined taxonomy.
- Create a new Taxonomy on the CMS python shell or use the [taxonomy-sample-data](https://github.com/open-craft/taxonomy-sample-data/tree/main).
- Go to the Taxonomy List page, you should be able to see the new taxonomy.
- Click on the tree dots on the new taxonomy card and on `Export`.
- Check the new Export modal.
- Click on `Export` on the Export modal with CSV format.
- Verify the downloaded file.
- Click on `Export` on the Export modal with JSON format.
- Verify the downloaded file.

**Other info**

- Internal ticket: [FAL-3528](https://tasks.opencraft.com/browse/FAL-3528).
- https://github.com/openedx/modular-learning/issues/105

KristinAoki committed:
feat: update list view to be table (#628)

jesperhodge committed:
fix  file upload thumbnails (#629)

Internal issue: https://2u-internal.atlassian.net/browse/TNL-11085

File upload gallery cards had distorted thumbnails. The goal is to fix this.
I centered the thumbnails instead of stretching them. I had to change the card layout a bit to do that, so in order to make everything look fine, I worked a bit on margins and paddings and font-size in order to bring this close to the figma mockups. It's not perfect because when you resize the browser window, the grid does some resizing that doesn't look as good as in figma, but I think it should be good enough for now.

renovate[bot] committed:
fix(deps): update dependency @edx/frontend-lib-content-components to v1.175.0 (#631)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

jesperhodge committed:
fix: info modal and list view thumbnails (#636)

ChrisChV committed:
feat: System-defined tooltip added

ChrisChV committed:
feat: Taxonomy card menu added. Export menu item added

PKulkoRaccoonGang committed:
fix: fixed sidebar margin top (#73) (#593)

ChrisChV committed:
feat: Modal for export taxonomy

jesperhodge committed:
chore: change files page title (#639)

Changed "Files & Uploads" page title to just "Files"

ChrisChV committed:
feat: Connect with export API

rpenido commented on src/taxonomy/api/hooks/api.js:
```diff
@@ -1,20 +1,53 @@
+      fileExtension = 'csv';
+      data = response.data;
+    }
+    downloadDataAsFile(data, contentType, `${name}.${fileExtension}`);
```
Calling the endpoint with the parameter `download=1` you could get the filename and extension from the backend using the `Content-Disposition` and simplify a bit here.

Another alternative is to leave the front end as it is and remove the `download` option to simplify the backend.

What do you think?

ChrisChV replied:
@rpenido We need to call the endpoint using Axios because the call needs to be authenticated. I have tried using `download=1`, but doesn't works, [here is the explanation](https://medium.com/@drevets/you-cant-prompt-a-file-download-with-the-content-disposition-header-using-axios-xhr-sorry-56577aa706d6). I've added that [as a comment](https://github.com/open-craft/frontend-app-course-authoring/pull/4/files/2070743175408ae9a844fa575b024ab88cec3605#diff-51e2c0d3363c83c5d167493ee852ee13f16564ad2c70189c74d66f8adaca8cd6R29).

> Another alternative is to leave the front end as it is and remove the download option to simplify the backend.

I think yes, we need to remove that because it will no longer be necessary, but it would be done in a follow-up task, I'm already running out of time here CC @bradenmacdonald

bradenmacdonald replied:
Doesn't the user have cookie authentication for the CMS domain? You don't want to call it as an API using Axios/fetch, you just want to send the user's browser to that URL. `window.location.href = '/path/to/download/api?download=1` - like that. Or even better, use `<a href="/path/to/api?download=1">Download</a>`. And @rpenido is right, you shouldn't need to specify the filename.

rpenido replied:
I also thought that the authentication would be a problem, but @bradenmacdonald is right. Making a request directly to the backend using the authenticated browser section worked (in my setup).

ChrisChV replied:
@rpenido How did you make the request? I have tried `window.location.href` way but doesn't work. CC @bradenmacdonald 


https://github.com/open-craft/frontend-app-course-authoring/assets/11827305/89eef993-e6ac-4ec4-bedc-5b971fb63622

bradenmacdonald replied:
@ChrisChV Weird. It works for me - I just went to http://studio.local.overhang.io:8001/api/content_tagging/v1/taxonomies/1/export/?output_format=json&download=1 (the tutor devstack equivalent URL) in my browser and it downloaded a file.

If you go to http://localhost:18010 are you logged in to Studio?

bradenmacdonald replied:
@ChrisChV  I think you are using an old version of openedx-learning, and you're missing this commit: https://github.com/openedx/openedx-learning/commit/d66d6294c1b27fbb285cda365802925f528bfe39

ChrisChV replied:
> If you go to http://localhost:18010/ are you logged in to Studio?

Yes, that is it. Now it works, thanks! That simplifies things

rpenido replied:
Your code worked here @ChrisChV.

ChrisChV committed:
Taxonomy list page [FC-0036] (#622)

bradenmacdonald commented:
> Is hidden for system-defined, because the export API doesn't allow export system-defined

Why does the API not allow exporting system-defined taxonomies? I discussed this with Axim and we decided that some power users may want to export system-level taxonomies, and there's no reason to prevent it.

Not in scope to fix now though.

ChrisChV committed:
test: Tests for API and selectors

feanil committed:
chore: Update to the new version of brand-openedx in the new scope. (#646)

Part of https://github.com/openedx/axim-engineering/issues/23

This updates the `@edx/brand` alias to point to the `brand-openedx` package at
the `openedx` scope. This does not impact imports because this package is used
via an alias.

jesperhodge committed:
chore: adjust renovate config (#637)

ChrisChV committed:
feat: Use windows.location.href to call the export endpoint
